### PR TITLE
Update modal styling for max width and z-index

### PR DESCRIPTION
**Pull request by dkasper-was-taken**
### Summary
Correct the maximum width for fixed size modal and add ability to set z-index additional layer.

After offline discussion with @neilpfeiffer we are looking to make these changes as well:
Removing responsive break-point behavior at this level, so each consuming implementation can handle responsiveness differently. The first two examples we have are within modal-manager and the alert/prompt.  Depending on the modal manager height, in some cases we don't want to go full screen in small form factor. Within the prompt we never want to go to full screen on small form factors.

Removing the theme-able variables for height, width, and margin. They are not being consumed and we do not want to forcibly set heights of all modal content across implementations.

Resolves: #1398, Resolves: #1323 

Thanks for contributing to Terra. 
@cerner/terra

### CODE REVIEW

File: packages/terra-modal/src/Modal.jsx [hunk 1]
```diff
@@ -4,6 +4,8 @@ import { Portal } from 'react-portal';
 import 'terra-base/lib/baseStyles';
 import ModalContent from './ModalContent';
 
+const zIndexes = ['6000', '8000'];
```
Comment (yuderekyu) [inline packages/terra-modal/src/Modal.jsx]: Just curious, how did we decide on these two z-index values?

Comment (dkasper-was-taken) [reply packages/terra-modal/src/Modal.jsx]: 6000 was the previous value, the popup is at 7000, so i went an additional layer to 8000.

Comment (mjhenkes) [reply packages/terra-modal/src/Modal.jsx]: I vote 9001, for no reason...

File: packages/terra-abstract-modal/src/AbstractModal.scss [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    max-height: calc(100% - 2 * var(--terra-abstract-modal-vertical-inset, 10px));
+    max-width: calc(100% - 2 * var(--terra-abstract-modal-horizontal-inset, 10px));
+    outline: none;
+    overflow: hidden;
+    position: fixed;
+    top: 50%;
+    transform: translate(-50%, -50%);
+
+    &.layer-6000 {
+      z-index: 6000 + 1;
```
Comment (tbiethman) [inline packages/terra-abstract-modal/src/AbstractModal.scss]: Any reason not to just make this 6001?

Comment (dkasper-was-taken) [reply packages/terra-abstract-modal/src/AbstractModal.scss]: Not especially, it was the pattern used in modal before, and kind of made a weird kind of sense to have 'layer-6000' be '6000'(+1). I'm fine updating it if people have an opinion either way.

File: packages/terra-abstract-modal/src/AbstractModal.jsx [hunk 3]
```diff
@@ -55,6 +51,10 @@ const propTypes = {
    * Role attribute on the modal dialog.
    */
   role: PropTypes.string,
+  /**
+   * Z-Index layer to apply to the ModalContent and ModalOverlay.
```
Comment (mjhenkes) [inline packages/terra-abstract-modal/src/AbstractModal.jsx]: document the available options here so they end up in the props table?

**Pull request merged**